internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -46,12 +47,11 @@ import (
 	"sigs.k8s.io/cluster-api/internal/hooks"
 	fakeruntimeclient "sigs.k8s.io/cluster-api/internal/runtime/client/fake"
 	"sigs.k8s.io/cluster-api/internal/test/builder"
-	. "sigs.k8s.io/cluster-api/internal/test/matchers"
 )
 
 var (
 	IgnoreNameGenerated = IgnorePaths{
-		{"metadata", "name"},
+		"metadata.name",
```

Comment:
Q: Should we use ObjectMeta.Name here instead to consistently use the go type format?

Reply:
The Go style format doesn't work with unstructured objects - even for metadata name as far as I can see.

Reply:
Ah alright. Does the "JSON"  format work in the other cases?

Reply:
It does - I've updated to make all the IgnorePaths in json format.